libevm/EVMC.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <evmjit.h>
+#include <libevm/VMFace.h>
+
+// FIXME: Move to EVM-C evm.h.
+typedef struct evm_instance* (*evm_create_fn)(void);
+
+namespace dev
+{
+namespace eth
+{
+class EVM;
+
+/// Translate the EVMSchedule to EVM-C revision.
+evm_revision toRevision(EVMSchedule const& _schedule);
+
+template<evm_create_fn createFn>
```

Comment:
So factory function has always the same signature, what's exactly the purpose of templatizing `EVMC` ?

You could just have it a simple subclass of `VMFace` and have various
```
std::uniqie_ptr<EVMC> createEvmJit();
std::uniqie_ptr<EVMC> createEvmWasm();
...
```
(then you'd need `#include <evmjit.h>` only in implementation, not in header)
or
```
std::uniqie_ptr<EVMC> createEvmC(evm_create_fn createFn);
```

Reply:
I wanted to do it without a template, but we have an issue with singletons at the moment. The template is a temporary workaround because it has `instance<hera>()` and `instance<evmjit>()`.

We have to rethink the lifetime of an VM instance. cpp-ethereum creates a new VM instance for every call.